src/character.cpp

```diff
@@ -3048,12 +3048,13 @@ SkillLevel &Character::get_skill_level_object( const skill_id &ident )
 
 int Character::get_skill_level( const skill_id &ident ) const
 {
-    return _skills->get_skill_level( ident );
+    return std::round( enchantment_cache->modify_value( ident, _skills->get_skill_level( ident ) ) );
```

Comment:
`_skills` is private to modify only, its data is exposed publicly through `get_all_skills()`, so you need to expand this check to more places.

Reply:
oh drat, that's a little awkward... it returns a ref. the easiest thing i can think of is to walk that back and make it a copy